examples/jsm/renderers/webgl-legacy/nodes/WebGLNodes.js

```diff
@@ -8,21 +8,23 @@ export const nodeFrame = new NodeFrame();
 
 Material.prototype.onBuild = function ( object, parameters, renderer ) {
 
-	if ( Array.isArray( object.material ) ) {
+	const material = this;
```

Comment:
@sunag I'm not sure this fix is compatible with #26841 though. When `onBuild()` is called in `WebGLRenderer`,  the material can't be an array of materials so using `const material = this;` and then `Array.isArray( object.material )` will always evaluate to `false`.

Reply:
If we can't fix this in `WebGLNodes`, I'll revert and add this to the renderer:
```js
if ( scene.overrideMaterial === null ) material.onBuild( object, parameters, _this );
```
That means however `Scene.overrideMaterial` can't be of type node material. But at least https://github.com/mrdoob/three.js/issues/27095 will be fixed.

Reply:
I think it would be good to fix this in `WebGLNodes` somehow since the assumption materials can be derived from the `object` reference isn't right. I've realized this does not only break `overrideMaterial` but shadow rendering when an objects with node materials are rendered with depth materials.

Reply:
I added a remaining reference [here](https://github.com/mrdoob/three.js/pull/27097/commits/a615adee23acc507d20e43903bf0d4e8e1173f11), it looks ok to me now using `overrideMaterial`.

Reply:
I have simplified the method a bit. The fiddle from https://github.com/mrdoob/three.js/issues/26811#issue-1904059962 still works and https://github.com/mrdoob/three.js/issues/27095 is also fixed.